File: Algorithms/sorting/selectionSort.py
def selectionSort(A, n):
   for i in range(n-1):
      imin = i
      for j in range(i+1, n):
         if A[j] < A[imin]:
            imin = j
      A[imin], A[i] = A[i], A[imin]

A = [7, 9, 5, 2, 3, 0, 100]

if __name__ == "__main__":
   selectionSort(A, 7)
   print(A)

File: Algorithms/sorting/selectionSort.cpp
/*Selection Sort(A, n) 
   {
      for i=0 to n-2
      {
        imin = i
         for j=i+1 to n-1
         {
          check(A[j] < A[imin])
            imin = j
         }
         swap(A[imin], A[j])
      }
   }
*/

#include <bits/stdc++.h>
using namespace std;

void selectionSort(int A[], int n)
{
   for (int i = 0; i < n - 1; i++)
   {
      int imin = i;
      for (int j = i + 1; j < n; j++)
      {
         if (A[j] < A[imin])
         {
            imin = j;
         }
      }
      int temp = A[i];
      A[i] = A[imin];
      A[imin] = temp;
   }
}

int main()
{
   int A[] = {7, 9, 3, 4, 0, 1};
   selectionSort(A, 6);
   for (int i = 0; i < 6; i++)
   {
      cout << A[i] << " ";
   }

   return 0;
}